autoload/ale/handlers.vim

```diff
@@ -220,3 +220,49 @@ function! ale#handlers#HandleStyleLintFormat(buffer, lines) abort
 
     return l:output
 endfunction
+
+function! ale#handlers#HandleGhcFormat(buffer, lines) abort
+    " Look for lines like the following.
+    "
+    " /dev/stdin:28:26: Not in scope: `>>>>>'
+     "Appoint/Lib.hs:8:1: warning:
+    let l:pattern = '^[^:]\+:\(\d\+\):\(\d\+\):\s\{-}\(warning\|error\)\(.\+\)$'
```

Comment:
So it turns out that the ghc 8.0 (the version I use locally) did in fact slightly tweak the output and so I've changed the handler and accompanying tests to ensure that this works with ghc 7.

I dropped the comment about matching `" /dev/stdin:28:26: Not in scope: `>>>>>'` because that's always been there [since this file was created](https://github.com/w0rp/ale/commit/57ef2c98339af5fd40e99e44c420815f9ed0805f)  but never would have matched the regex